kb/2023-08-11/using_netapp_third_party_storage.md

```diff
@@ -0,0 +1,154 @@
+---
+title: Using NetApp Storage on Harvester
+description: Installation procedure for NetApp Astra Trident CSI Driver
+slug: install_netapp_trident_csi
+authors:
+  - name: Jeff Radick
+    title: Staff Software Engineer
+tags: [harvester]
+hide_table_of_contents: false
+---
+
+Instructions for installing the Netapp Astra Trident CSI driver
+into a Harvester cluster.
+
+This enables use of NetApp storage systems to store storage volumes
+usable by virtual machines running in Harvester.
+
+The NetApp storage will be an option in addition to the normal
+Longhorn storage; it will not replace Longhorn.
+Virtual machine images will still be stored using Longhorn.
+
+This has been tested with Harvester 1.2.0 and Trident v23.07.0.
+
+This procedure only works to access storage via iSCSI, not NFS.
+
+Note also that 3rd party storage classes (including those based on Trident)
+can only be used for non-boot volumes of Harvester VMs.
+
+== Detailed Instructions
+
+We are assuming that before beginning this procedure,
+a Harvester cluster and a NetApp ONTAP storage system
+are both installed and configured for use.
+
+Follow all of the following steps, but ONLY on the initial cluster node.
+
+The last step (setting up multipathd) should be done on all nodes,
+AFTER the initial node has had Trident installed.
+
+1. Read the NetApp Astra Trident documentation
+       https://docs.netapp.com/us-en/trident/
+       https://docs.netapp.com/us-en/trident/trident-get-started/kubernetes-deploy-operator.html
+   We are describing the "manual deployment" procedure here.
+2. Log into the Harvester cluster in one of the nodes.
```

Comment:
users need to login to one of the `master` nodes or download the kubeconfig and run the commands locally.

Reply:
I think in my major revision I've described the method for downloading and using the kubeconfig locally, along with the procedure for using helm.  Your feedback on those changes will be appreciated.

Reply:
I didn't say anything about doing stuff on a `master` node though.  Perhaps I should?

Reply:
I actually did put in a few words about this later on.